hostfs/hostfs.c

```diff
@@ -910,7 +910,6 @@ static int _fs_dup(
     ECHECK((tret = myst_tcall(SYS_dup, params)));
 
     new_file->fd = tret;
-    ret = tret;
```

Comment:
ramfs and ext2fs `fs_dup` returns 0 on success. The fd allocation and assignment is done in `myst_syscall_dup`.